app/routes/admin.py

```
@@ -4479,39 +4484,66 @@ def student_detail_public(actor_public_id):
 @admin_bp.route('/student/<int:seat_id>/adjust-hall-pass-entitlements', methods=['POST'])
 @admin_required
-@feat_shell("FEAT-STOR-001")
 def adjust_hall_pass_entitlements(seat_id):
     """Grant or remove hall-pass entitlements for a student."""
-    student = db.session.get(Seat, seat_id)
-    if not student:
+    try:
+        canonical_context = getattr(g, "canonical_context", None)
+        if not canonical_context:
+            abort(403)
+    except Exception:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)